frontend/src/core/user/components/UserControls.js

```diff
@@ -0,0 +1,55 @@
+/* @flow */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import AppSwitcher from './AppSwitcher';
+import SignIn from './SignIn';
+import SignOut from './SignOut';
+import UserAutoUpdater from './UserAutoUpdater';
+import { actions, NAME } from '..';
+
+import type { UserState } from 'core/user';
+
+
+type Props = {|
+    router: Object,
+    user: UserState,
+|};
+
+type InternalProps = {
+    ...Props,
+    dispatch: Function,
+};
+
+
+export class UserControlsBase extends React.Component<InternalProps> {
+    getUserData = () => {
+        this.props.dispatch(actions.get());
+    }
+
+    render() {
+        const { router, user } = this.props;
+
+        return <div className='user-controls'>
+            <UserAutoUpdater getUserData={ this.getUserData } />
+
+            { user.isAuthenticated ?
+                <SignOut dispatch={ this.props.dispatch } url={ user.signOutURL } /> :
```

Comment:
I haven't seen `dispatch` being passed around like this. Am I doing something wrong?

Reply:
Yes, `dispatch` should never be passed around. Instead you should create a new method here that uses `dispatch` and pass that method down. I can take it up from here though, let me review by pushing some commits. :sweat_smile: